tests/integ/modin/frame/test_quantile.py

```diff
@@ -111,3 +129,18 @@ def test_quantile_datetime_negative():
     snow_df = pd.DataFrame(TEST_QUANTILE_DATA)
     with pytest.raises(NotImplementedError):
         snow_df.quantile(numeric_only=False)
+
+
+@pytest.mark.xfail(
```

Comment:
do you have some more context on this test? Is the problem here the `to_pandas()` emitting a lot of queries?

Reply:
This is a modified version of `series/test_quantile.py::test_quantile_large`, which this PR addresses by removing UNION ALLs. That optimization is only applicable in the single-column case, so performing a quantile() operation with many qs on a multi-column frame will run into the same high query counts as before.

Reply:
ok, thanks for explaining. Do you have an idea whether we could do some similar to approach to get rid off the high union count there as well?

Reply:
Possibly, but the simplest thing I can think of would be just trading off UNIONs for JOINs -- we could use this UNPIVOT method to compute a quantile column for every column in the input frame, then JOIN all those columns together.

This may be worth it if len(q) >> len(columns), though it might take some trial and error for us to figure out an appropriate threshold.